src/v/kafka/server/handlers/fetch.cc

```diff
@@ -554,7 +604,7 @@ void op_context::create_response_placeholders() {
                 .partition_index = fp.partition,
                 .error_code = error_code::none,
                 .high_watermark = fp.high_watermark,
-                .last_stable_offset = fp.high_watermark,
+                .last_stable_offset = fp.last_stable_offset,
```

Comment:
is this fixing a bug unrelated to supporting transactions? if so, please note it in the commit message.